h2o-core/src/main/java/hex/grid/filter/PermutationFilterFunction.java

```diff
@@ -1,13 +1,9 @@
 package hex.grid.filter;
 
+import hex.Model;
 
-import java.util.Map;
+import java.util.function.Predicate;
 
-/**
- * By using {@link FunctionalInterface} it is possible to define lambda functions of a corresponding type. 
- * Note: interface with {@link FunctionalInterface} will not provide full functional support ( e.g. no `compose` and `andThen` methods)
- */
-@FunctionalInterface
-public interface PermutationFilterFunction {
-  Boolean apply(Map<String, Object> permutation);
+public interface PermutationFilterFunction<MP extends Model.Parameters>
+        extends FilterFunction<MP>, Predicate<MP>, Activatable {
```

Comment:
would it make sense to have a default implementations for `apply`, `reset` and `activate`?
```
default Boolean apply(MP mp) { return test(mp); }
default void activate() {}
default void reset() {}
```
This way, you can still make this a `FunctionInterface`.